addons_tests/test_module/models/hop_dong.py

```diff
@@ -1,19 +1,74 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from datetime import *
+from odoo.exceptions import ValidationError
+
+
+def get_signed_date():
+    signed_date = datetime.now() + timedelta(weeks=1)
+    if signed_date.weekday() == 5:
+        signed_date += timedelta(days=2)
+    elif signed_date.weekday() == 6:
+        signed_date += timedelta(days=1)
+    return signed_date
 
 
 class hop_dong(models.Model):
     _name = 'hop.dong'
     _description = 'hop dong'
 
-    name = fields.Char(string='ten')
-    start_date = fields.Datetime(string='ngay bat dau')
-    end_date = fields.Datetime(string='ngay ket thuc')
-    contract_type = fields.Char(string='loai hop dong')
-    signed_date = fields.Datetime(string='ngay ky')
-    salary_rack = fields.Float(string='luong ngach bac')
-    efficiency_wage = fields.Float(string='luong hieu qua')
+    name = fields.Char(string='Name')
+    start_date = fields.Datetime(string='Start Date: ', default=fields.Datetime.now, format='%Y-%m-%d')
+    end_date = fields.Datetime(string='End Date: ')
+    contract_type = fields.Char(string='Contract Type: ')
+    signed_date = fields.Datetime(string='Signed Date: ', default=get_signed_date())
+    salary_rack = fields.Float(string='Standard Salary: ')
+    efficiency_wage = fields.Float(string='Efficiency Wage: ')
     status = fields.Selection([
-        ('draft', 'ban nhap'),
-        ('ongoing', 'dang dien ra'),
-        ('expired', 'het han')],
-        string='trang thai', default='draft')
+        ('new', 'New'),
+        ('running', 'Running'),
+        ('expired', 'Expired'),
+        ('pause', 'Pause')],
+        string='Status', default='new')
+
+    employee_id = fields.Many2one('hr.employee', string='Employee')
+
+    employee_name = fields.Char(string='Employee Name', required=True, compute='_compute_employee_name')
+
+    employee_role = fields.Char(string='Employee Role')
+
+    employee_job = fields.Char(string='Employee Job')
+
+    employee_level = fields.Char(string='Employee Level')
+
+    total_salary = fields.Float(string='Total Salary', compute='_compute_total_salary')
+
+    @api.depends('salary_rack', 'efficiency_wage')
+    def _compute_total_salary(self):
+        for record in self:
```

Comment:
tại sao chỗ này lại dùng for, không dùng for có được không

Reply:
for ở đây dùng cho việc tính toán riêng các bản ghi với nhau nên không dùng for là không được ạ